src/Scrima.OData.Swashbuckle/OpenApiQueryTypesFilter.cs

```diff
@@ -1,31 +1,32 @@
 using System.Linq;
-using Microsoft.OpenApi.Any;
-using Microsoft.OpenApi.Models;
+using Microsoft.OpenApi;
 using Swashbuckle.AspNetCore.SwaggerGen;
 
 namespace Scrima.OData.Swashbuckle;
 
 internal class OpenApiQueryTypesFilter : IDocumentFilter, ISchemaFilter
 {
-    public void Apply(OpenApiSchema schema, SchemaFilterContext context)
+    public void Apply(IOpenApiSchema schema, SchemaFilterContext context)
     {
-            if (context.Type.IsODataQuery())
-            {
-                schema.Extensions["odata-schema"] = new OpenApiBoolean(true);
-            }
+        if (context.Type.IsODataQuery() && schema.Extensions != null)
+        {
+            schema.Extensions["odata-schema"] = new JsonNodeExtension(true);
         }
-
+    }
+    
     public void Apply(OpenApiDocument swaggerDoc, DocumentFilterContext context)
     {
-            var schemasToRemove = swaggerDoc.Components.Schemas
-                .Where(s => s.Value.Extensions.TryGetValue("odata-schema", out var value) &&
-                            value is OpenApiBoolean {Value: true})
-                .Select(s => s.Key)
-                .ToList();
-            
-            foreach (var schemaKey in schemasToRemove)
-            {
-                swaggerDoc.Components.Schemas.Remove(schemaKey);
-            }
+        var schemasToRemove = swaggerDoc.Components?.Schemas
+            ?.Where(s => 
+                s.Value.Extensions != null &&
+                s.Value.Extensions.TryGetValue("odata-schema", out var value) &&
+                value is { })
+            .Select(s => s.Key)
+            .ToList();
+
+        foreach (var schemaKey in schemasToRemove)
```

Comment:
not sure, but now it seems that `schemasToRemove` can be `null`? maybe better to check for it before the foreach